Skewed Arrows, 0.7.X support and more!

EdwhakKB opened the pull request:
This is a big work inside modcharting tools (adding more modifiers, a tutorial for lua and a skew sprites [simulates 3d somehow])
Huge thanks to Glowsoony for the help with 0.7.x and Skewed Sprite fix!

EdwhakKB committed:
Psych 0.7 fixes

fixes for 0.7 and a few stuff for modchart editor for begginers

EdwhakKB committed:
Psych 0.7 fixes

Added fixes for 0.7 version of Psych engine / improved a bit the modchart editor with some new stuff for begginers

EdwhakKB committed:
oops

i did it wrong lmao

EdwhakKB committed:
Better stuff for modchart editor ofc

Since it looks better like this

EdwhakKB committed:
yes

needed

EdwhakKB committed:
Just to fix some stuff here

since i think this can fix a lot of stuff but if you want and can, remove it lol

EdwhakKB committed:
New Stuff!

SkewedArrows, fixed stuff, added 0.7.x support, new modifiers etc etc

moxie-coder reviewed (approved):
seems good so far

EdwhakKB committed:
oh god i forgot personal build stuff

srry

Rorutop commented:
Yo gonna use this! But what about specifying the version? Since there are 2 version inside the source folder, I'm using psych 6.3

moxie-coder commented:
> Yo gonna use this! But what about specifying the version? Since there are 2 version inside the source folder, I'm using psych 6.3

use the 0.6.3 folder then

Vortex2Oblivion reviewed (approved):
lgtm

glowsoony commented:
**Please note that this PR is a test! Skewed notes are still being tested on to see the limits it may have! so far its already looking complete from the code me and Edwhak have seen.** _Please know 0.7.x is still experimental_ and I am testing that Version while Edwhak is testing 0.6.3 and so far skewed notes are a major success.

EdwhakKB committed:
Support for modifier explain

Added support for 0.6.X and 0.7.X vers, also fixed some modifiers and added difficulty modcharts

EdwhakKB committed:
OOPS

MINIMAL SPELLING MISTAKE OH NO

EdwhakKB committed:
Fixed some issues here

I forgot some 0.7.X stuff so...

EdwhakKB committed:
HotFix

Added a fix for all crash issues

EdwhakKB committed:
BRO

EVERY 5 SECONDS A NEW BUG LMAO

EdwhakKB committed:
Fixed Modchart Difficulty

4 hours of PAIN with glowsoony to try fix this shit lmao

EdwhakKB committed:
HaxeLib Update

Asked by Vortex since man didn't wanted use copy and paste code edition lmao, if there is any bug with ur leather engine tell me

EdwhakKB committed:
OOPS

I FORGOT MODCHART DIFFICULTY 0.7 SUPPORT XDDDDD

EdwhakKB committed:
BRO

at least its way better ig

EdwhakKB committed:
fine, i'll see any other bugs myself

i expected 1 or 2, not 50

EdwhakKB committed:
...

BRO

EdwhakKB committed:
Fixed Stuff

I hope i finally fixed this, i just want to rest now...

EdwhakKB committed:
Fixed Update

Finally working all 0.7 stuff!

EdwhakKB committed:
fixed "setSubMod" c++ bugs

Some people was complaining about this issue so i just fixed it ig lmao have fun

EdwhakKB committed:
forgot XD

forgot

EdwhakKB committed:
Improved some modfiers

Added scroll support for Skew modifier, making able to choose if its forced to skew at that dir, or if it depends on upscroll or downscroll, ah yeah notes mods too lmao

EdwhakKB committed:
TOTAL LEATHER SUPPORT

FINALLY BRO LMAO

EdwhakKB committed:
._.XD

.-.XD

EdwhakKB committed:
finally fixed support

idk why it was making issues and shit but oh well who cares anymore

EdwhakKB committed:
OOPS

XD

EdwhakKB committed:
Bro is kinda blind

BRO, I ALMOST THOUGHT I KILLED SOMETHING THEN I SAW YOU WAS ULTRA BLIND LMAOOOO

EdwhakKB committed:
some new stuff

i orderer some stuff and also fixed a LOT of math (no funny besier system for u rorutob lol)

EdwhakKB committed:
some things i forgot to fix

srry

EdwhakKB committed:
some extra modifier ig

the arrow path modifier its still experimental so do not even try use it!

EdwhakKB committed:
to avoid crash

yeah to avoid crash lol

EdwhakKB committed:
PATH MODIFIER FIX

FINALLY FINALLY AAAAAAAAAA

EdwhakKB committed:
Reverse subVariables

since i think its better use them like different modifiers

EdwhakKB committed:
joe tio

XDDDD

EdwhakKB committed:
AAAAA

MALDITASEA

2JENO commented:
hey sustain note is invisible help

EdwhakKB committed:
some new modifiers ig?

next update will be 0.7.2 support

Vortex2Oblivion committed:
fix leather engine support once and for all (plus some other funnies)

Vortex2Oblivion committed:
i forgor

EdwhakKB committed:
Merge pull request #2 from Vortex2Oblivion/main

Fix leather engine once and for all (Thanks vortex lol)

Vortex2Oblivion committed:
i forgot about html5 lols

Vortex2Oblivion committed:
Update ModchartFuncs.hx

EdwhakKB committed:
Merge pull request #3 from Vortex2Oblivion/main

bro is a genious

EdwhakKB committed:
0.7.3 and more

just fixed some math and the improvement of 0.7.3 (with help of Glowsoony)

EdwhakKB committed:
glow xuxesumare

bro deleted leather anticrash :skull:

Vortex2Oblivion committed:
Update ModchartEditorState.hx

Vortex2Oblivion committed:
Update ModchartFile.hx

EdwhakKB committed:
Merge pull request #5 from Vortex2Oblivion/main

told you, fue glow

EdwhakKB committed:
hello

hello

EdwhakKB committed:
Merge branch 'main' of https://github.com/EdwhakKB/FNF-Modcharting-Tools

EdwhakKB committed:
yes

yes

Vortex2Oblivion commented:
Haxelibs can have their own import.hx files iirc

EdwhakKB commented:
e

EdwhakKB committed:
huh?

huh

EdwhakKB committed:
im sorry i forgot

sorry

EdwhakKB committed:
a

glowsoony committed:
Just change Psych into it's subversion instead of Seprate

glowsoony committed:
READ PRESET CHANGES LMAO

glowsoony committed:
Lol

glowsoony committed:
Update README.md

glowsoony committed:
Sorry Ed For the Spaces

glowsoony committed:
HSCRIPT!

glowsoony committed:
Update ModchartFuncs.hx

glowsoony committed:
Forgot that SCE Needs this FR

glowsoony committed:
Added the stupid thing because I won't deal with the opponentMode strum error's any more.

glowsoony committed:
Oops

glowsoony committed:
Update NoteMovement.hx

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
InitMode for HScript README.md

glowsoony committed:
I GIVE YOU LIFE ONCE MORE! (remove if not working)

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
Lol

glowsoony committed:
Flx Sound

glowsoony committed:
Just in case.

glowsoony committed:
WHATTTTTTTTTTTTTTT

glowsoony committed:
oopsi

glowsoony committed:
OOPSI

glowsoony committed:
Whatever

glowsoony committed:
LOL

glowsoony committed:
Update README.md / Everyone please read this

glowsoony committed:
Update README.md / Everyone please read this

glowsoony committed:
Im testing this lmao for downscroll for SCE and 0.7 really

glowsoony committed:
0.7

glowsoony committed:
what 💀💀💀💀💀

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
Suffix For Paths Update

glowsoony committed:
CLEAR MEMORY

glowsoony committed:
Update ModchartFile.hx

EdwhakKB committed:
now?

glow, stfu, you broke it yet again

EdwhakKB committed:
FUCK

FUCK

EdwhakKB committed:
...

...

glowsoony committed:
Even though suffixForPath isn't the problem, I am still updating, SUFFIX FOR PATH IS NOT AT FAULT AT ALL!!!!

glowsoony committed:
Quant for SCE

glowsoony committed:
Quant

glowsoony committed:
Since SCE will update with these changes I am forced to change this, anyone with SCE WILL NOT BE ABLE TO EXPORT UNTIL THE NEW VERSION IS OUT!

glowsoony committed:
Delete source/managers directory

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
Update ModchartEditorState.hx

glowsoony committed:
Update ModchartEditorState.hx

glowsoony committed:
Upcoming SCE update

glowsoony committed:
Just for SCE

glowsoony committed:
a

glowsoony committed:
Testing HScriptImproved with modcharts

glowsoony committed:
Lets hope it works lol

glowsoony committed:
I forgot mine already does File.getContent()

glowsoony committed:
PLEASE WORK PLEASE PLEASE PLEASE

glowsoony committed:
:skull:

glowsoony committed:
Update ModchartFile.hx

glowsoony committed:
My bad

glowsoony committed:
Create Dynamic Creation

glowsoony committed:
Dynamic Parent for Sets

glowsoony committed:
erm

EdwhakKB committed:
Added bezier path

funky

glowsoony committed:
EDWHAKKKKKKKKKKKKK!!!!!!!! 😭😭😭😭😭😭

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
Fix this thing

glowsoony committed:
Edwhak te voy a matar

glowsoony committed:
Update PlayfieldRenderer.hx

glowsoony committed:
Only for SCE unless someone makes support for other engines

glowsoony committed:
GPU Render MT fix by 2JENO 🗣️🗣️🗣️🗣️🗣️🗣️🗣️🔥🔥🔥🔥🔥🔥🔥

glowsoony committed:
Update README.md 2JENO credits added 🗣️🔥🔥🔥

glowsoony committed:
Update README.md

Vortex2Oblivion committed:
yes i do like to build the game

EdwhakKB committed:
Merge pull request #8 from Vortex2Oblivion/main

cool

glowsoony committed:
Both of you forgot this function as-well 😔😔😔😔😔😔😔

glowsoony committed:
HITMANS X HSI

glowsoony committed:
I think this is fine.

glowsoony committed:
SCE's Needed change for changing notes (touches nothing else) 😭💀

Es para SCE edwhak nada mas

glowsoony committed:
My bad, forgot about others 😭🙏

glowsoony committed:
Don't forget you own engine!!! (ed)

glowsoony committed:
Volume

Vortex2Oblivion committed:
use flxmath.fastsin / flxmath.fastcos

glowsoony committed:
Update ModchartFile.hx (Because the update is not done for HaxeUI and SCE EXP is already updated)

PLEASE TELL ME IF YOU CANT EXPORT DIRECTLY ON DISCORD OR MAKE PULL REQUEST (SCEN'T PEOPLE)

EdwhakKB committed:
LMAO

fixed

glowsoony committed:
Update ModTable.hx

glowsoony committed:
Update ModchartFuncs.hx

Vortex2Oblivion committed:
Merge branch 'EdwhakKB:main' into main

Vortex2Oblivion committed:
Update import.hx

Vortex2Oblivion committed:
Update ModchartEditorState.hx

EdwhakKB committed:
Merge pull request #10 from Vortex2Oblivion/main

noice

Vortex2Oblivion committed:
Update ModchartEditorState.hx

EdwhakKB committed:
Merge pull request #11 from Vortex2Oblivion/main

stopid

LuckCTE commented:
I have a problem. Error with compiling.

`C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/Modifier.hx:13: characters 8-17 : Type not found : PlayState
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/Modifier.hx:14: characters 8-12 : Type not found : Note
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/PlayfieldRenderer.hx:27: characters 8-17 : Type not found : PlayState
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/PlayfieldRenderer.hx:28: characters 8-12 : Type not found : Note
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/PlayfieldRenderer.hx:42: characters 24-33 : Type not found : StrumNote
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/NoteMovement.hx:13: characters 8-17 : Type not found : PlayState
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/NoteMovement.hx:14: characters 8-12 : Type not found : Note
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/ModchartUtil.hx:14: characters 8-17 : Type not found : PlayState
C:/HaxeToolkit/haxe/lib/fnf-modcharting-tools/git/source/modcharting/ModchartUtil.hx:15: characters 8-12 : Type not found : Note
source/psychlua/FunkinLua.hx:56: characters 17-30 : Expected }`

roma-perec-bp commented:
Uhh my sustains are invisible

glowsoony commented:
_**Some things to know. (For Psych ONLY)**_

1. If your on 0.7.x (0.7 or higher) turn off GPU Caching, this is not a bug, what happens is that GPU Caching destroys the image and re builds it, storing the memory on GPU instead of CPU, turn it off.

2. Check what version you put in your project.xml! define the version as `<define name="PSYCHVERSION" value=""/>` the value is the version your on so it would be "0.6.3" or whatever version your on! also be sure to define the variable "PSYCH" as `<define name="PSYCH" />` If you define psych  and define the psychversion than it will look for what things to import, this is important! As what imports is depending both on the engine (defined as PSYCH) and the (PSYCHVERSION defined as whatever version you put).

roma-perec-bp commented:
wait, variable backupGpu doesn't work at all?

glowsoony commented:
In the editor if it's not visible Im check that out since it should be, but if neither in playstate nor modcharteditor then it's a bug (in the case that GPU rendering is on) but if it's not working in the editor try with it off and if it works then I'll check that out

roma-perec-bp commented:
firstly i had them invisible in playstate, but when i entered editor they became visible even in playstate

glowsoony commented:
If they became visible after entering the editor then I see, it is the fact GPU Rendering is turned on, the Editor has support for turning it off but by default PlayState won't clearly since the support is only for the editor.

roma-perec-bp commented:
weird

roma-perec-bp commented:
also question i'm kinda dumb lol, how skew modifier works lol, i think i don't understand smth

glowsoony commented:
SkewModifier really doesn't work sometimes or Idk if it works cause I find that sometimes its depending on a few things, but Skew works by applying a _skew_ to the receptors such as looking like its bending forward

EdwhakKB commented:
> also question i'm kinda dumb lol, how skew modifier works lol, i think i don't understand smth

Hey, im Mt creator and i'll tell, Skew modifier its a modifier that uses subValues (X.Y,XD,YD) so in order to properly make it work you should first enable the base modifier

(have in mind it should already been created)
lua should be:

set(beat, [[
1,skewMod
]])

in editor you go to events, pick the skew modifier and add value 1 (add time to 0 to instantly do it since "SET" in editor doesn't work properly yet)

then you are able to modify sub values

glowsoony committed:
ModchartingTools WITHOUT SCE

glowsoony committed:
Update PlayfieldRenderer.hx

Vortex2Oblivion committed:
someone did an oopsie

EdwhakKB committed:
Merge pull request #13 from Vortex2Oblivion/main

leather shit LOL

ThatOneSillyGuyDon commented:
been looking through this and looks pretty cool, now my main question is if someone's gonna be able to implement support for those that uses multikeys on this?

Been trying to do it myself but had no luck unfortunately.

EdwhakKB commented:
> been looking through this and looks pretty cool, now my main question is if someone's gonna be able to implement support for those that uses multikeys on this?
> 
> Been trying to do it myself but had no luck unfortunately.

Multikeys as psych engine multikeys or an engine that actually has multikeys by default (Leather)?

If psych then i can tell that im currently on a whole MT rework due some lag issues, but this update will be mainly and exclusively for psych engine, why? well, i see that having Leather engine support and other engines such as Codename or Andromeda (which has it's own template) is quite useless, so MT will be Psych engine exclusive after this update, so far to add multikeys (psych) you need modify noteMovement.hx and the way to load the strums onto a dynamic way
